File: Domain/Entities/Operation.cs
using Domain.Enums;

namespace Domain.Entities;

public sealed class Operation : BaseEntity
{
    public string Name { get; set; }
    public OperationType OperationType { get; set; }
    public float Value { get; set; }
    public DateTimeOffset Date { get; set; }
    
    public Guid SeasonId { get; set; }
    public Season Season { get; set; }
    public Guid FieldId { get; set; }
    public Field Field { get; set; }

    private Operation(string name, OperationType operationType, float value, Guid seasonId, Guid fieldId, DateTimeOffset date)
    {
        Name = name;
        OperationType = operationType;
        Value = value;
        SeasonId = seasonId;
        FieldId = fieldId;
        Date = date;
    }

    public static Operation Create(string name, OperationType operationType, float value, Guid seasonId, Guid fieldId, DateTimeOffset date)
        => new(name, operationType, value, seasonId, fieldId, date);

    public void Update(string name, OperationType operationType, float value)
    {
        Name = name;
        OperationType = operationType;
        Value = value;
    }
}

File: client-app/src/app/models/operation.ts

export interface Operation {
    id: string,
    name: string,
    operationType: string,
    value: number,
    seasonId?: string,
    fieldId?: string,
    fieldNumber?: number,
    date: Date | null
}


export class Operation implements Operation {
    constructor(init: Operation) {
        this.id = init.id;
        this.name = init.name;
        this.operationType = init.operationType;
        this.value = init.value;
        this.date = init.date;
    }

    id: string;
    name: string;
    operationType: string;
    value: number;
    date: Date | null;
}

export interface CreateOperationResponse {
    operationId: string
}